shell/browser/api/electron_api_web_contents.cc

```diff
@@ -1086,6 +1095,59 @@ void WebContents::Invoke(bool internal,
                  std::move(callback), internal, channel, std::move(arguments));
 }
 
+void WebContents::ReceivePostMessage(const std::string& channel,
+                                     blink::TransferableMessage message) {
+  v8::HandleScope handle_scope(isolate());
+  auto wrapped_ports =
+      MessagePort::EntanglePorts(isolate(), std::move(message.ports));
+  v8::Local<v8::Value> message_value =
+      electron::DeserializeV8Value(isolate(), message);
+  EmitWithSender("-ipc-ports", bindings_.dispatch_context(), InvokeCallback(),
+                 false, channel, message_value, std::move(wrapped_ports));
+}
+
+void WebContents::PostMessage(gin::Arguments* args) {
+  std::string channel;
```

Comment:
Why are we pulling the arguments out manually here instead of letting gin to the work?

```
void WebContents::PostMessage(std::string channel, v8::Local<v8::Value> message_value, ...)
```

Reply:
done